### PR TITLE
Comments: Fix comments on post types that don't support Jetpack comment forms.

### PR DESCRIPTION
**Pull request by dereksmart**
Fixes #62646

#### Changes proposed in this Pull Request:

- Smarter nonce checks hooked onto pre_comment_on_post

#### Jetpack product discussion

p1649892098202479-slack-CBG1CP4EN

#### Does this pull request change what data or activity we track or use?

No

#### Testing instructions:

* Activate Jetpack Comments 
* Add a product via Woo
* Try to leave a review

### CODE REVIEW

File: projects/plugins/jetpack/modules/comments/comments.php [hunk 1]
```diff
@@ -536,6 +536,11 @@ public function watch_comment_parent() {
 	public function pre_comment_on_post() {
 		$post_array = stripslashes_deep( $_POST );
 
+		/** This filter is documented in modules/comments/comments.php */
+		if ( ! apply_filters( 'jetpack_comment_form_enabled_for_' . get_post_type( $post_array['comment_post_ID'] ), true ) ) {
```
Comment (dereksmart) [inline projects/plugins/jetpack/modules/comments/comments.php:540]: This won't work as-is. Realized there's this same check at the bottom of this method that's die'ing out intentionally. We'll need another solution.

Comment (coder-karen) [reply projects/plugins/jetpack/modules/comments/comments.php:540]: Worked on a separate solution over here -  #23945 - would that do that job?

Comment (dereksmart) [reply projects/plugins/jetpack/modules/comments/comments.php:540]: Great! Thanks for taking it 👍

**Comment by dereksmart** [comment]
Will be handled in #23945

**Pull request closed without merging**